meilisearch-http/src/routes/indexes/documents.rs

```diff
@@ -227,26 +230,54 @@ async fn document_addition(
 
     let (uuid, mut update_file) = index_scheduler.create_update_file()?;
 
-    // TODO: this can be slow, maybe we should spawn a thread? But the payload isn't Send+Sync :weary:
-    // push the entire stream into a `Vec`.
-    // If someone sends us a never ending stream we're going to block the thread.
-    // TODO: Maybe we should write it to a file to reduce the RAM consumption
-    // and then reread it to convert it to obkv?
-    let mut buffer = Vec::new();
+    let temp_file = match NamedTempFile::new() {
```

Comment:
There are a lot of drawbacks to using a `NamedTempFile`, and I'm not sure I see the point. We're not persisting the file or anything; it's only used as a temporary buffer while we wait to write the data in another file :thinking: 
Why don't we use a simpler [TempFile](https://docs.rs/tempfile/latest/tempfile/fn.tempfile.html)?

Reply:
I also tried `tempfile()`  and it doesn't seem to work well for this scenario. To persist the byte stream of the payload to the tempfile, and read it again, the `seek` method must be used after the entire byte stream has been transferred, which doesn't seem like a good choice especially if the payload is particularly large. Using `tempfile()`and `seek()` introduces a new problem, when `read_json` fails if it tries to deserialize to seq first, it currently uses `serde_json::from_reader` to deserialize, and in this case the contents of tempfile are not available.

Reply:
Huum, not sure I see the issue with using `seek` on a very large file; it's made for that and pretty much answers in constant time, right?

> when read_json fails if it tries to deserialize to seq first, it currently uses serde_json::from_reader to deserialize, and in this case the contents of tempfile are not available.

And I'm also not sure I understood this part; sorry :grimacing: 
`tempfile()` returns a real `File`. Thus you can read and write in it or give it to `serde_json::from_reader`?
How are the contents of tempfile not available?

Reply:
I used the same code as you yesterday, but when I tested it, I found many problems during deserialization. When debugging, it was found that the file read during deserialization was empty. I tried it again today and it works perfectly. Thank you for your help.👍